packages/jest-cli/src/jest.js

```diff
@@ -290,175 +48,40 @@ const runCLI = (
 
   readConfig(argv, root)
     .then(config => {
-      if (argv.debug) {
-        /* $FlowFixMe */
-        const testFramework = require(config.testRunner);
-        pipe.write('jest version = ' + VERSION + '\n');
-        pipe.write('test framework = ' + testFramework.name + '\n');
-        pipe.write('config = ' + JSON.stringify(config, null, '  ') + '\n');
-      }
-      if (argv.watch || argv.watchAll) {
-        setWatchMode(argv, argv.watch ? 'watch' : 'watchAll', {
-          pattern: argv._,
-        });
-
-        let currentPattern = '';
-        let hasSnapshotFailure = false;
-        let isEnteringPattern = false;
-        let isRunning = false;
-        let testWatcher;
-        let timer;
-        let displayHelp = true;
-
-        const writeCurrentPattern = () => {
-          clearLine(pipe);
-          pipe.write(chalk.dim(' pattern \u203A ') + currentPattern);
-        };
-
-        const startRun = (overrideConfig: Object = {}) => {
-          if (isRunning) {
-            return null;
-          }
+      createDirectory(config.cacheDirectory);
+      const jestHasteMap = Runtime.createHasteMap(config, {
+        console: new Console(pipe, pipe),
+        maxWorkers: getMaxWorkers(argv),
+        resetCache: !config.cache,
+        watch: config.watch,
+      });
 
-          testWatcher = new TestWatcher({isWatchMode: true});
-          pipe.write(CLEAR);
+      return jestHasteMap.build().then(
+        hasteMap => ({
```

Comment:
we could potentially move `createHasteContext` into a utility function inside of jest-cli and do this:

```
const createHasteContext = (config, {hasteFS, moduleMap}) => {
  hasteFS,
  resolver: Runtime.createResolver(config, moduleMap),
});
```

this way we can reuse this "map" operation of the code in two places. I think it makes sense, this may change in the future and I'd like to reuse code if possible :)